File: src/my_first_package/my_first_package/tests_publisher_subscriber/publisher_test.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class PublisherTest(Node):
    def __init__(self, node_name):
        super().__init__(node_name)
        # Parámetros: tipo de dato, nombre topic, queue size del topic
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.timer_ = self.create_timer(0.5, self.publish)
        self.get_logger().info("Publisher test started")
    
    def publish(self):
        msg = String()
        msg.data = "Publisher test"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = PublisherTest("publisher_test")
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_first_cpp_pkg/src/tests_topics/publisher_test.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class PublisherTest : public rclcpp::Node
{
    public:
    PublisherTest() : Node("my_first_node") 
    {
        m_publisher = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        m_timer = this->create_wall_timer(std::chrono::milliseconds(500), 
                                    std::bind(&PublisherTest::publish, this));
        RCLCPP_INFO(this->get_logger(), "Publisher started");                        
    }
    private:
        void publish() { 
            auto msg = example_interfaces::msg::String();
            msg.data = std::string("Cpp node publisher");
            m_publisher->publish(msg); 
        }
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr m_publisher;
        rclcpp::TimerBase::SharedPtr m_timer;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PublisherTest>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}